oauth2_http/java/com/google/auth/oauth2/ComputeEngineCredentials.java

```diff
@@ -102,13 +107,22 @@
    *
    * @param transportFactory HTTP transport factory, creates the transport used to get access
    *     tokens.
+   * @param scopes Scope strings for the APIs to be called. May be null or an empty collection.
    */
-  private ComputeEngineCredentials(HttpTransportFactory transportFactory) {
+  private ComputeEngineCredentials(
+      HttpTransportFactory transportFactory, Collection<String> scopes) {
     this.transportFactory =
         firstNonNull(
             transportFactory,
             getFromServiceLoader(HttpTransportFactory.class, OAuth2Utils.HTTP_TRANSPORT_FACTORY));
     this.transportFactoryClassName = this.transportFactory.getClass().getName();
+    this.scopes = (scopes == null) ? ImmutableSet.<String>of() : ImmutableSet.copyOf(scopes);
```

Comment:
what if the collection is not empty but a string in the collection is null or empty?

Reply:
modified the code to remove null and empty string from scopes, and updated the test